From 6a250fe2fc5db7ae2ba2355a1e00743e306854fc Mon Sep 17 00:00:00 2001
From: auto commit <unknown@unknown>
Date: Fri, 24 Oct 2008 12:23:13 -0700
Subject: [PATCH] cavium: Hook I2C support into the Kconfig

Add the I2C Kconfig/Makefle options in for the new Cavium I2C files

Signed-off-by: Tomaso Paoletti <tpaoletti@caviumnetworks.com>
Signed-off-by: Paul Gortmaker <Paul.Gortmaker@windriver.com>
---
 drivers/i2c/busses/Kconfig  |    9 +++++++++
 drivers/i2c/busses/Makefile |    7 +++++++
 2 files changed, 16 insertions(+), 0 deletions(-)

diff --git a/drivers/i2c/busses/Kconfig b/drivers/i2c/busses/Kconfig
index eb94378..77d17fd 100644
--- a/drivers/i2c/busses/Kconfig
+++ b/drivers/i2c/busses/Kconfig
@@ -716,4 +716,13 @@ config SCx200_ACB
 	  This support is also available as a module.  If so, the module
 	  will be called scx200_acb.
 
+config I2C_OCTEON_TWSI
+	tristate "Octeon TWSI-I2C"
+	depends on I2C
+	help
+	  Octeon TWSI-I2C
+
+	  This driver can also be built as a module.  If so, the module
+	  will be called i2c-octeon-twsi.
+
 endmenu
diff --git a/drivers/i2c/busses/Makefile b/drivers/i2c/busses/Makefile
index d4dc22d..104d0a5 100644
--- a/drivers/i2c/busses/Makefile
+++ b/drivers/i2c/busses/Makefile
@@ -46,6 +46,7 @@ obj-$(CONFIG_I2C_SH7760)	+= i2c-sh7760.o
 obj-$(CONFIG_I2C_SH_MOBILE)	+= i2c-sh_mobile.o
 obj-$(CONFIG_I2C_SIMTEC)	+= i2c-simtec.o
 obj-$(CONFIG_I2C_VERSATILE)	+= i2c-versatile.o
+obj-$(CONFIG_I2C_OCTEON_TWSI)	+= i2c-octeon_twsi.o
 
 # External I2C/SMBus adapter drivers
 obj-$(CONFIG_I2C_PARPORT)	+= i2c-parport.o
@@ -71,3 +72,9 @@ obj-$(CONFIG_I2C_OMAP)          += i2c-omap.o
 ifeq ($(CONFIG_I2C_DEBUG_BUS),y)
 EXTRA_CFLAGS += -DDEBUG
 endif
+
+ifdef CONFIG_I2C_OCTEON_TWSI
+OCTEON_ROOT = $(srctree)/arch/mips/cavium-octeon
+
+CFLAGS_i2c-octeon_twsi.o	= -I$(OCTEON_ROOT)/executive
+endif
-- 
1.5.5.1

